metis-enrichment/metis-enrichment-service/src/main/java/eu/europeana/enrichment/service/dao/WikidataAccessDao.java

```diff
@@ -197,25 +190,24 @@ private Resource getModelFromSPARQL(String uri) {
    * @return true if the retrieved resource is duplicated
    */
   private boolean isDuplicate(Resource resource) {
-    return (resource.hasProperty(OWL.sameAs) && !resource.hasProperty(RDFS.label));
+    return (resource!= null && resource.hasProperty(OWL.sameAs) && !resource.hasProperty(RDFS.label));
```

Comment:
Wait are you sure about this change?
The isDuplicate is called with a negation, so !isDuplicate inside the getModelFromSPARQL method.
Which means that if resource is null this line here evaluates to false and the !isDuplicate evaluates to true, therefore returning a null value. That null value is passed in the transform method which with through NPE in that scenario..

Reply:
yes ... resource is dupplicated is exists at all (not null), has same as and has no label.
A resource that doesn't exists (is null) cannot be considered to be dupplicated.
This method is used to check if the resource URI is a kind of redirect (dupplicated resource) in wikidata.

Reply:
I get that but the if it is null the transform method will throw an NPE on line 222